test/gracefulstop_test.go

```diff
@@ -42,7 +42,9 @@ type delayListener struct {
 	closeCalled  chan struct{}
 	acceptCalled chan struct{}
 	allowCloseCh chan struct{}
+	mu           sync.Mutex
 	dialed       bool
+	closed       bool
```

Comment:
@pvsravani where and how are we using `closed`. Why is it needed when switching to stub server?

Reply:
The closed field ensures the close() method is idempotent. It prevents multiple calls to close() from closing the listener again, which could cause errors.